tests/integration/examples/utils.py

```diff
@@ -111,24 +120,23 @@ def run_example(
 
     yield example, runs
 
+    # Cleanup registered pipelines so they don't cause trouble in future
+    # example runs
+    for pipeline in client.depaginate(list_method=client.list_pipelines):
+        if pipeline.id not in existing_pipeline_ids:
+            client.delete_pipeline(pipeline.id)
+
     cleanup_docker = request.config.getoption("cleanup_docker", False)
 
     if cleanup_docker:
         # Clean up more expensive resources like docker containers, volumes and
         # images, if any were created.
+        image_names = set()
+        for build in client.depaginate(list_method=client.list_builds):
+            if build.id in existing_build_ids:
+                continue
 
-        active_stack = Client().active_stack_model
-        if StackComponentType.CONTAINER_REGISTRY in active_stack.components:
-            container_registry = active_stack.components[
-                StackComponentType.CONTAINER_REGISTRY
-            ][0]
-            image_name = f"{container_registry.configuration['uri']}/zenml:{pipeline_name}"
-        else:
-            # For orchestrators that don't need a container registry (e.g. local
-            # Docker orchestrator and local Airflow orchestrator) and for step
-            # operators, we leverage the default convention used to name the
-            # image `zenml:<pipeline_name>`.
-            image_name = f"zenml:{pipeline_name}"
+            image_names.update(item.image for item in build.images.values())
```

Comment:
So good to see the build power in action here !

Same as before though, can you determine the list of builds and images from the list of pipeline runs returned by wait_and_validate_pipeline_run ? That should help focus this deadly laser beam a bit.